pkg/util/cluster_files/cluster_files.go

```
@@ -0,0 +1,100 @@
+// a local path to the file
+func Get(k8sClient kubernetes.Interface, urlAddress string) (pathStr string, err error) {
+	klog.V(log.DEBUG).Infof("reading cluster_file: %s", urlAddress)
+	parsedUrl, err := url.Parse(urlAddress)
+	if err != nil {
+		klog.Error(err)
+		return "", errors.Wrapf(err, "Trying to read cluster file %q", urlAddress)
+	}
+
+	namespace := parsedUrl.Host
+	pathContainerObject, pathFile := path.Split(parsedUrl.Path)
+	pathContainerObject = strings.Trim(pathContainerObject, "/")
+
+	if pathContainerObject == "" || pathFile == "" {
+		return "", MalformedURL
+	}
+
+	var data []byte
+
+	switch parsedUrl.Scheme {
+	case "file":
+		return parsedUrl.Path, nil
+
+	case "secret":
+		secret, err := k8sClient.CoreV1().Secrets(namespace).Get(pathContainerObject, metav1.GetOptions{})
+		if err != nil {
+			return "", errors.Wrapf(err, "Error reading secret %q from namespace %q", pathContainerObject, namespace)
+		}
+		var ok bool
+		data, ok = secret.Data[pathFile]
+		if !ok {
+			return "", FileContentNotFound
+		}
+
+	case "configmap":
+		configMap, err := k8sClient.CoreV1().ConfigMaps(namespace).Get(pathContainerObject, metav1.GetOptions{})
+		if err != nil {
+			return "", errors.Wrapf(err, "Error reading configmap %q from namespace %q", pathContainerObject, namespace)
+		}
+		var ok bool
+		data, ok = configMap.BinaryData[pathFile]
+		if !ok {
+			dataStr, ok := configMap.Data[pathFile]
+			if !ok {
+				return "", FileContentNotFound
+			}
+
+			data = []byte(dataStr)
+		}
+
+	default:
+		return "", UnsupportedSchema
+	}
+
+	return storeToDisk(pathContainerObject, parsedUrl, data)
+}
+
+func storeToDisk(pathContainerObject string, parsedUrl *url.URL, data []byte) (string, error) {
+	storageDirectory, err := ioutil.TempDir("", "cluster_files")
+	if err != nil {
+		klog.Error(err)
+		return "", errors.Wrap(err, "Error creating cluster_files directory")
+	}
+
+	dir := path.Join(storageDirectory, pathContainerObject)
+	_ = os.MkdirAll(dir, 0700)
```

Should we just ignore this error?

---

I thought it would return error if the directory already existed, but it looks like it wont. Let me add error handling